### PR TITLE
Report module multimedia

### PR DESCRIPTION
**Pull request by nickpell**
@orangejenny cc: @NoahCarnahan 

screen shot 2015-11-17 at 1 55 31 pm

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/report-module.js [hunk 1]
```diff
@@ -287,18 +298,19 @@ var ReportModule = (function () {
...
-                        reports: JSON.stringify(_.map(self.reports(), function (r) { return r.toJSON(); }))
+                        reports: JSON.stringify(_.map(self.reports(), function (r) { return r.toJSON(); })),
+                        multimediaUpdate: JSON.stringify(self.multimediaUpdate())
                     }
                 });
             }
         });
 
-        var changeSaveButton = function () {
+        self.changeSaveButton = function () {
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/report-module.js]: curious, what's the reason for this change?

File: corehq/apps/app_manager/static/app_manager/js/report-module.js [hunk 2]
```diff
@@ -287,18 +298,19 @@ var ReportModule = (function () {
                     data: {
                         name: JSON.stringify(self.moduleName),
                         module_filter: self.moduleFilter,
-                        reports: JSON.stringify(_.map(self.reports(), function (r) { return r.toJSON(); }))
+                        reports: JSON.stringify(_.map(self.reports(), function (r) { return r.toJSON(); })),
+                        multimediaUpdate: JSON.stringify(self.multimediaUpdate())
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/report-module.js]: picky, but I'd just call this `multimedia`, everything in this hash is an update

File: corehq/apps/app_manager/static/app_manager/js/report-module.js (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/report-module.js]: nm, hadn't finished reading

File: corehq/apps/app_manager/templates/app_manager/partials/report_configs_js.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            moduleName: {{ module.name|JSON }},
+            moduleFilter: "{{ module.module_filter|escapejs }}",
+            availableReports: {{ all_reports|JSON }}, // structure for all reports
+            currentReports: {{ current_reports|JSON }}, // config data for app reports
+            saveURL: saveURL,
+            lang: {{ lang|JSON }},
+            menuImage: menuImage,
+            menuAudio: menuAudio
+        });
+        ko.applyBindings(reportModule, $("#save-button")[0]);
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/report_configs_js.html]: super picky, but you could just use `document.getElementById` instead of going through jQuery

File: corehq/apps/app_manager/views/modules.py [hunk 4]
```diff
@@ -194,6 +202,35 @@ def _report_to_config(report):
         'all_reports': [_report_to_config(r) for r in all_reports],
         'current_reports': [r.to_json() for r in module.report_configs],
         'warnings': warnings,
+        'multimedia': {
```
Comment (orangejenny) [inline corehq/apps/app_manager/views/modules.py]: Most of this is identical to https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/views/view_generic.py#L172-L186 - any objection to DRYing it up?

**Comment by orangejenny** [comment]
Looks good, just a few minor comments.

File: corehq/apps/app_manager/views/modules.py (hunk 4 above)
Comment (nickpell) [inline corehq/apps/app_manager/views/modules.py]: turns out this was already being added to the context

File: corehq/apps/app_manager/views/modules.py [hunk 5]
```diff
@@ -26,6 +26,14 @@
     prefix_usercase_properties, commtrack_ledger_sections)
 
 from corehq.apps.fixtures.models import FixtureDataType
+from corehq.apps.hqmedia.controller import (
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F401 'MultimediaAudioUploadController' imported but unused

Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F401 'MultimediaImageUploadController' imported but unused

File: corehq/apps/app_manager/views/modules.py [hunk 6]
```diff
@@ -26,6 +26,14 @@
...
+    MultimediaAudioUploadController,
+    MultimediaImageUploadController,
+)
+from corehq.apps.hqmedia.views import (
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F401 'ProcessAudioFileUploadView' imported but unused

Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F401 'ProcessImageFileUploadView' imported but unused

**Comment by orangejenny** [comment]
looks good, just waiting on tests
